sycl/source/detail/graph_impl.cpp

```diff
@@ -356,6 +392,23 @@ void graph_impl::makeEdge(std::shared_ptr<node_impl> Src,
   removeRoot(Dest); // remove receiver from root node list
 }
 
+void graph_impl::duplicateGraph(const graph_impl &GraphImpl) {
+  std::map<node_impl *, std::shared_ptr<node_impl>> NodesMaps;
+  for (auto &Root : GraphImpl.MRoots) {
+    auto CpyRoot = Root->duplicateNodeAndSuccessors(NodesMaps);
+    this->addRoot(CpyRoot);
+  }
+
+  for (auto &Queue : GraphImpl.MRecordingQueues) {
+    MRecordingQueues.insert(Queue);
+  }
```

Comment:
Should we be copying this? It's not clear to me that a queue would conceptually be recording to a copied graph. If so, then a comment about why `MRecordingQueues` needs to be copied would be useful

Reply:
I agree that conceptually we do not need to keep track of recorded queues. But I added it because the implementation releases the in-recording queues when the associated exec_graph is deleted.
So we want to use a duplicated graph in exec_graph we have to keep track of these queues or change the release strategy.